streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -550,6 +578,14 @@ public StateStore getStateStore(final String name) {
         return store instanceof SessionStore ? (SessionStore<K, V>) store : null;
     }
 
+    /**
+     * Processor topology getter
+     * @return the processor topology
+     */
+    public ProcessorTopology getProcessorTopology() {
```

Comment:
As above: `ProcessorTopology` is an internal class and should not be exposed to the user.

Reply:
I haven't been around here long enough to be confident about this, but here goes...

It looks like those tests are making assertions about the structure of the runtime topology that results from different DSL operations. To test that, it should be sufficient to directly call the relevant `build` operations to get a `ProcessorTopology` from a `Topology`. It doesn't seem to need the TopologyTestDriver to be involved.

Nevertheless, this illustrates the tension between an internal test driver and an external one. IMO, this class should be only an external test driver, so if there's some stuff that we need for internal testing purposes, then we need to keep the internal test driver around.

So I guess my suggestions are:
1. try to rewrite the those tests to avoid using a topology test driver at all
2. if that fails, just use the internal test driver instead

Reply:
I reverted back this back to `KStreamTestDriver` to avoid having to expose the internal class